src/main.cpp

```diff
@@ -717,18 +779,11 @@ int main(int argc, char* argv[])
   std::string filename;
   std::string program;
 
-  if (script.empty())
+  if (!args.filename.empty())
   {
-    // Script file
-    if (argv[optind] == nullptr)
-    {
-      LOG(ERROR) << "USAGE: filename or -e 'program' required.";
-      return 1;
-    }
```

Comment:
More argument parsing that has been moved into `parse_args`, which either populates `args.script` (for `-e`) or `args.filename` (for a positional parameter).